Automattic/jetpack
# Boost: Fix sending multiple tracks when speed score changes

## Proposed changes:

* Fix multiple tracks events when the Speed Score changed popup shows up;
* Fix sending tracks event even when the popup was hidden.

![CleanShot 2024-12-20 at 13 16 45](https://github.com/user-attachments/assets/e6f695cb-5171-40a6-9256-6d36c1534c97)

### Other information:


## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

The only was able to test this is by rigging the speed scores via the Boost companion plugin.

- Open network tab in console and filter for XHR requests (you can also query for `admin-ajax.php` requests);
- Setup Boost free;
- If the speed score changed popup shows, make sure only one track event is shown;
- If it shows, close it with the X;
- Now without refreshing the page, refresh the speed scores. There shouldn't be a tracks request for the "score changed" popup.
- Now refresh the page so the popup shows up - there should be a tracks request;
- Dismiss the popup and refresh the scores - there should be no tracks requests for the popup as it was dismissed.
